### DISCUSSION TITLE
How to prevent refresh component with wire:click

### DISCUSSION THREAD

**Discussion by Krendzel** [livewire/livewire#8993] [1. Help]
Hello, 
I have very basic layout where on left side i have list with users and on right side i have details for given user after click on user in list. 

blade: 
```blade
<x-layouts.app>
    <div class="flex gap-5">
        <div class="w-6/12">
            <livewire:users.list-users />
        </div>
        <div class="w-6/12 ">
            <livewire:users.show-user />
        </div>
    </div>
</x-layouts.app>
```

In table (list-users) i have this code: 

```php
 @foreach($users as $user)
  <tr wire:click="selectUser({{$user}})" wire:key="{{ $user->id }}" class="border-b hover:bg-gray-100">
      <th scope="row" class="px-4 py-3 font-medium text-gray-900 whitespace-nowrap">{{ $user->username }}</th>
      <td class="px-4 py-3">{{ $user->name }}</td>
      <td class="px-4 py-3">{{ $user->email }}</td>
      <td class="text-xs px-4 py-3">{{ $user->title }}</td>
      <td class="px-4 py-3">{{ $user->created_at->format('d/m/Y') }}</td>
  </tr>
  @endforeach
```

selectUser function:
```php
public function selectUser(User $user): void
    {
        $this->dispatch('userSelected', user: $user);
    }
```

In User details (show-user) i have function that listening for this event: 
```php
 #[On('userSelected')]
    public function userSelected(User $user): void
    {
        $this->user = $user;
        $this->form->setUser($user);
    }
```
This code works just fine but i noticed that when i triger "wire:click" i see two requests, one refreshing list, second request is with that event (OK)

![image](https://github.com/user-attachments/assets/91aa5d38-086b-4a5c-99a3-aff3a92c4759)

How i can prevent refreshing component when i trigger wire:click? I don't see any reason why wire:click should refresh component with wire:click.

**Comment by alan-su-jetfuel** [comment]
Usually use id.

```php
wire:click="userSelected({{ $user->id }})"
```

```php
public function userSelected($user_id): void
{
    $this->user = User::findOrFaile($user_id);
    $this->form->setUser($this->user);
}
```

**Comment by N1ebieski** [comment]
Maybe a [renderless attribute](https://livewire.laravel.com/docs/actions#skipping-re-renders)?
